interface/wx/maskededit.h

```diff
@@ -0,0 +1,935 @@
+
+        Resets all parameters, deleting previous ones.
+    */
+    wxMaskedEditParams();
+
+    /**
+        Resets all per-field parameters, deleting previous ones, and
+        setting room for @a numFields fields.
+    */
+    void ResetFields(size_t numFields);
+
+    /**
+        The string passed to the masked edit control with wxMaskedEdit::SetMask().
+    */
+    wxString mask;
+
+    /**
+        Colours used for different cases. The whole control is coloured.
+    */
+    wxMaskedEditColours colours;
+
+    /**
+        The container for each field flags.
+    */
+    wxVector<wxEditFieldFlags> fieldsFlags;
+
+    /**
+        The container for each field test function.
+
+        @see wxMaskedEdit::SetFieldFunction()
+    */
+    wxVector<wxMaskedFieldFunc*> fieldsFuncs;
+
+    /**
+        Container for the parameters for each wxMaskedFieldFunc
+
+        @see wxMaskedEdit::SetFieldFunction()
+    */
+    wxVector<void*> fieldFuncsParams;
+
+    /**
+        Pointer to the function for testing the whole control.
+
+        @see wxMaskedEdit::SetControlFunction()
+    */
+    wxMaskedFunc* controlFunc;
+
+    /**
+        Parameters to pass to the function for testing the whole control.
+
+        @see wxMaskedEdit::SetControlFunction()
+    */
+    void* controlFuncParams;
+};
+
+/**
+    @class wxMaskedEdit
+
+    This is the base class for all masked edit controls. Here is where most
+    of their features are implemented.
+
+    A masked edit control is like a form, with boxes where you write the data.
+    You may have a 4-cells field to write a year, a 2-cells field to write a day
+    of the month, a group of 4 fields, each of them is 3-cells long to write an
+    IP v4 address, etc.
+
+    Outside fields you can set "literals". They are characters that may help
+    the user to type the data. This literals are fixed, the user can't change
+    them.
+
+    Each cell accepts a predefined type of character. These types can be digits,
+    letters, digits or letters, any printable character, etc. All cells in a
+    field must accept the same type; the field must be "homogeneous". You can't
+    mix in the same field some cells accepting only digits and other cells
+    accepting only letters.
+
+    This per-cell definition, literals included, is called a "mask".
+
+    Examples. With this code:
+    @code
+    wxMaskedEditText *ctr1, *ctr2;
+    ctr1 = new wxMaskedEditText(this, ID1);
+    ctr1->SetMask("99#.99#.99#.99#");
+    ctr1->SetAllFieldsFlags(wxEditFieldFlags(wxALIGN_RIGHT));
+    ctr2 = new wxMaskedEditText(this, ID2);
+    ctr2->SetMask("####/9#/9#");
+    ctr2->SetAllFieldsFlags(wxEditFieldFlags(wxALIGN_RIGHT));
+    @endcode
+    you could create two masked controls, and set an IP mask (4 fields) and
+    a yyyy/mm/dd date mask (3 fields). All fields are also set to be right
+    aligned. Their appearance, empty and with some characters typed, may be
+    like these:
+    @image html maskededit.png
```

Comment:
The PNG file is missing from the PR.

Reply:
File is now pushed to the PR.

I will mark this as Resolved, since GitHub will not mark it "Outdated".